lib/ecto/migration/runner.ex

```diff
@@ -264,6 +266,32 @@ defmodule Ecto.Migration.Runner do
 
   ## Helpers
 
+  defp perform_operation(repo, module, operation) do
+    # TODO: Test adapter can never trigger this case, so how do we test it?
+    in_transaction? =
+      try do
+        repo.in_transaction?()
+      rescue
+        _ -> false
+      end
```

Comment:
So we shouldn't need the try/catch if we implement `in_transaction?` in the test adapter. `if function_exported?(repo, :in_transaction?, 0) and repo.in_transaction?()` below should be all we need. :)

Reply:
Ah, yeah sorry that slipped my mind. Sorry!